packages/@aws-cdk/aws-s3/lib/auto-delete-objects-handler/index.ts

```diff
@@ -0,0 +1,43 @@
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  // eslint-disable-next-line @typescript-eslint/no-require-imports, import/no-extraneous-dependencies
+  const s3 = new (require('aws-sdk').S3)();
```

Comment:
Why not initialize this out of the `handler` scope into the global one? This would avoid passing of the instance to every function and also the usage of `any` type for the s3 client.

Reply:
Ah right, this is a remnant of how I was mocking the s3 client previously - I'll go fix this.

Reply:
You could do something like:
```ts
import * as AWS from 'aws-sdk';

const s3 = new AWS.S3();
```

Reply:
So, for reasons I can't ascertain, when I hoist the `const s3 = new (require(...` line to the file's outer scope, or even perform a regular import like you suggested, then jest stops mocking the module properly, causing all my tests to run actual aws-sdk API calls (the first error it runs into a `NoSuchBucket` exception since the tests use dummy bucket names).

One thing to keep in mind is that the @aws-cdk/aws-s3 package doesn't actually have aws-sdk as a dependency. (That makes the above behavior odd - but the entire repo shares a single node_modules directory which does have aws-sdk so it's not that surprising).

Based on that, unless there is a way to fix this import behavior, I'll just keep passing the s3 client around. At the end of the day I think this is OK - I've tested that it works, and I'm not worried about any performance implications of instantiating the client inside the method since the "create" and "delete" operations only get once per bucket-and-custom-resource pair. But if you have any tips to fix this that is cool too. :)

Reply:
The jest mocks might not be working in this new scenario because of how the actual module was imported. Previously, since the module was imported inside the handler, the `jest.mock` call could be placed in the `describe` block.

Now that the module is imported at the file level, the `jest.mock` should also be done at the top of the tests file (since the actual module gets imported as soon as you import the handler file).

Please give it a try once. If it works, great! Otherwise you can keep this in the current form.

Reply:
Ah, that made it work! Thank you for teaching something new. :) I did have to move the mock above the handler import in order for it to fully work, but I think this is ok.